[tanboonkhong] iP

tanboonkhong opened the pull request:
Certainly! Here's a sample PR description using GitHub Flavored Markdown (GFMD):

---

## Duke Task Manager :crown:

> "The best way to get something done is to begin." - **Unknown**

Introducing **Duke Task Manager** - your one-stop solution for task management. It's:

- :pencil2: Text-based
- :rocket: Super easy to learn
- :zap: Extremely fast to use

### Getting Started

1. :arrow_down: [Download it from here](#).
2. :arrow_forward: Double-click to start.
3. :memo: Add your tasks.
4. :sunglasses: Let Duke manage your tasks for you.

**Bonus:** It's absolutely **FREE**!

### Features

- [ ] Task Management
- [ ] Deadline Management (Coming Soon)
- [ ] Reminders (Coming Soon)

For the developers out there, you can use Duke to practice your Java skills. Here's a sneak peek into the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

tanboonkhong committed:
Implement Level 0. Rename, Greet, Exit

tanboonkhong committed:
Rename the chatbot to Max

tanboonkhong committed:
Implement Level 1. Echo

tanboonkhong committed:
Implement Level 2. Add, List

tanboonkhong committed:
Implement Level 3. Mark as Done

tanboonkhong committed:
Implement Level 4. ToDos, Events, Deadlines

tanboonkhong committed:
Implement Automated Text UI Testing

tanboonkhong committed:
Implement Level 5. Handle Errors

tanboonkhong committed:
Implement Level 6. Delete

tanboonkhong committed:
Implement List using Collections

tanboonkhong committed:
Implement Level 7. Save

tanboonkhong committed:
Merge branch-level-7

tanboonkhong committed:
Implement Level 8. Dates and Times

tanboonkhong committed:
Merge branch-Level-8

tanboonkhong committed:
Implement A-MoreOOP: Use More OOP

tanboonkhong committed:
Merge pull request #1 from tanboonkhong/branch-A-MoreOOP

Implement A-MoreOOP: Use More OOP

tanboonkhong committed:
Add package duke

tanboonkhong committed:
Merge branch 'add-gradle-support'

tanboonkhong committed:
Add Gradle support

tanboonkhong committed:
Add JUnit test cases

tanboonkhong committed:
Add test cases

tanboonkhong committed:
Add JavaDoc

tanboonkhong committed:
Follow Coding Standard

tanboonkhong committed:
Implement Level-9

tanboonkhong committed:
Update follow coding standard

tanboonkhong committed:
Add JavaDoc for Level-9

tanboonkhong committed:
Update JavaDoc

tanboonkhong committed:
Update JavaDoc

tanboonkhong committed:
Revert "Update JavaDoc"

This reverts commit 207e1b67031bf9b00a3780cd6e095808579ca2fe.

tanboonkhong committed:
Merge branch 'branch-A-JavaDoc'

tanboonkhong committed:
Update Coding Standard

tanboonkhong committed:
Merge branch 'branch-A-CodingStandard'

tanboonkhong committed:
Revert "Add JavaDoc for Level-9"

This reverts commit 4b7b878899261795a10d5cd516c8657ecbbed6e1.

tanboonkhong committed:
Revert "Revert "Add JavaDoc for Level-9""

This reverts commit 276fb6b618d71411c666def5c65e613b81b3994f.

tanboonkhong committed:
Update Level-9

tanboonkhong committed:
Merge branch 'branch-Level-9'

pangyyen commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+        if (!Files.exists(path.getParent())) {
+            try {
+                Files.createDirectories(path.getParent());
+            } catch (Exception e) {
+                e.printStackTrace();
```
Would it be beneficial to handle the exception more gracefully by throwing a custom exception, like 'DukeException', or displaying a more user friendly message?

e.g.
```java
try {
    Files.createDirectories(path.getParent());
} catch (IOException e) {
    System.err.println("An I/O error occurred while creating directories: " + e.getMessage());
} catch (SecurityException e) {
    System.err.println("A security error occurred while creating directories: " + e.getMessage());
}
```

KamiliArsyad replied:
I agree with @pangyyen here.

pangyyen commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+        }
+        try {
+            Files.write(Paths.get(filePath), saveLines);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Same as the suggestion mentioned above in this file (Storage.java)

example:
```java
catch (IOException e) {
    throw new DukeException("Error reading/writing the file: " + e.getMessage());
}
```

pangyyen reviewed (commented):
Overall, the code is well-structured and organised, and adheres to OOP principles such as encapsulation, Inheritance, composition, readability and documentation.

KamiliArsyad commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,117 @@
+            break;
+        default:
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
```
This seem like a rather tedious and repetitive process. Perhaps it's better to refactor some methods out?

KamiliArsyad commented on src/main/java/duke/TaskManager.java:
```diff
@@ -0,0 +1,95 @@
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskManager {
+    private static final String FILE_PATH = "./data/duke.txt";
+    private List<Task> tasks;
+
+    public TaskManager() {
+        this.tasks = new ArrayList<>();
+        loadTasks();
+    }
+
+    public void addTask(Task task) {
```
Will be better with javadoc written here, too!

KamiliArsyad reviewed (commented):
Really hard to find any coding standard-related issues so I think overall really well done on the code. LGTM

tanboonkhong committed:
Implement Level 10

tanboonkhong committed:
Implement Level 10

tanboonkhong committed:
Implement Level 10

tanboonkhong committed:
Implement Level 10 GUI

tanboonkhong committed:
Merge branch 'branch-Level-10'

tanboonkhong committed:
Update UI of the App

The App is not returning the message correctly.

It should return the message based on the user's input.

The getResponse is modified so that it returns the string instead
of printing the string.

tanboonkhong committed:
Merge branch 'branch-Level-10'

tanboonkhong committed:
Add assertions to Duke class

Assertions is introduced in the Duke class to ensure certain conditions
hold true during the runtime.

- Ensure tasks are initialized after loading from storage.
- Confirm that user and duke text labels are initialized.

We can be more confident about the correctness of the code using
these assertions.

tanboonkhong committed:
Merge branch 'branch-A-Assertions'

tanboonkhong committed:
Add JavaDoc comments to Duke class methods

Added comprehensive JavaDoc comments to the methods in Duke class
Improved code readability and maintainability for the code

tanboonkhong committed:
Merge pull request #2 from tanboonkhong/A-CodeQuality

Add JavaDoc comments to Duke class methods

tanboonkhong committed:
Update Parser.java

tanboonkhong committed:
Merge pull request #3 from tanboonkhong/branch-A-CodeQuality

Update Parser.java

tanboonkhong committed:
Update Parser.java

tanboonkhong committed:
Merge pull request #4 from tanboonkhong/branch-A-Assertions

Update Parser.java

tanboonkhong committed:
Implemented B-DoWithinPeriodTasks

tanboonkhong committed:
Merge branch 'B-Period'

tanboonkhong committed:
Added User Guide for the project

tanboonkhong committed:
Merge pull request #6 from tanboonkhong/branch-A-UserGuide

Added User Guide for the project

tanboonkhong committed:
Add Ui.png

tanboonkhong committed:
Merge pull request #7 from tanboonkhong/branch-A-UserGuide

Add Ui.png

tanboonkhong committed:
Updated Application Name

tanboonkhong committed:
Merge pull request #8 from tanboonkhong/branch-A-UserGuide

Updated Application Name

tanboonkhong committed:
Modify gradle

tanboonkhong committed:
Update Parser.java

tanboonkhong committed:
Merge pull request #9 from tanboonkhong/branch-gradle

Branch gradle
